packages/synapse-react-client/src/components/CardContainerLogic/CardContainerLogic.tsx

```diff
@@ -218,11 +217,13 @@ export const CardContainerLogic = (props: CardContainerLogicProps) => {
           props.noContentPlaceholderType ?? NoContentPlaceholderType.STATIC
         }
       >
-        {sortConfig && <QuerySortSelector sortConfig={sortConfig} />}
-        <CardContainer {...props} />
-        <QueryContextConsumer>
-          {queryContext => <ErrorBanner error={queryContext?.error} />}
-        </QueryContextConsumer>
+        <QueryWrapperErrorBoundary>
+          {sortConfig && <QuerySortSelector sortConfig={sortConfig} />}
+          <RowSetView
+            cardConfiguration={props}
+            initialLimit={props.initialLimit}
+          />
```

Comment:
Instead of rendering `CardContainer` directly, render `RowSetView`, which will internally handle fetching paginated query data, rendering the cards, and rendering pagination controls.